tests/core/test_proxy.py

```diff
@@ -44,3 +65,63 @@ def test_no_proxy(self):
         env.pop("https_proxy", None)
         env.pop("HTTP_PROXY", None)
         env.pop("HTTPS_PROXY", None)
+
+
+class CustomAgentTransaction(AgentTransaction):
+
+    def on_response(self, response):
+        super(CustomAgentTransaction, self).on_response(response)
+        self._test.stop()
```

Comment:
Also, if you could add a comment as to what `test` should be, maybe pointing here (that would be great): http://www.tornadoweb.org/en/stable/testing.html